File: backend/models/mcc_mnc.go
package models

import (
	"time"
)

type MccMnc struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	Mcc         string    `json:"mcc" gorm:"size:10;not null;index"`
	Mnc         string    `json:"mnc" gorm:"size:10;not null;index"`
	Iso         string    `json:"iso" gorm:"size:10;not null;index"`
	Country     string    `json:"country" gorm:"size:100;not null;index"`
	CountryCode string    `json:"country_code" gorm:"size:10;not null"`
	Network     string    `json:"network" gorm:"size:200;not null;index"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type MccMncCreateRequest struct {
	Mcc         string `json:"mcc" validate:"required,min=1,max=10"`
	Mnc         string `json:"mnc" validate:"required,min=1,max=10"`
	Iso         string `json:"iso" validate:"required,min=1,max=10"`
	Country     string `json:"country" validate:"required,min=1,max=100"`
	CountryCode string `json:"country_code" validate:"required,min=1,max=10"`
	Network     string `json:"network" validate:"required,min=1,max=200"`
}

type MccMncUpdateRequest struct {
	Mcc         string `json:"mcc" validate:"required,min=1,max=10"`
	Mnc         string `json:"mnc" validate:"required,min=1,max=10"`
	Iso         string `json:"iso" validate:"required,min=1,max=10"`
	Country     string `json:"country" validate:"required,min=1,max=100"`
	CountryCode string `json:"country_code" validate:"required,min=1,max=10"`
	Network     string `json:"network" validate:"required,min=1,max=200"`
}

type MccMncResponse struct {
	ID          uint      `json:"id"`
	Mcc         string    `json:"mcc"`
	Mnc         string    `json:"mnc"`
	Iso         string    `json:"iso"`
	Country     string    `json:"country"`
	CountryCode string    `json:"country_code"`
	Network     string    `json:"network"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type MccMncFilterOptions struct {
	Countries []string `json:"countries"`
	Networks  []string `json:"networks"`
	Isos      []string `json:"isos"`
}

File: frontend/src/services/mcc-mnc.ts
import { apiClient } from './api-client';

export interface MccMnc {
  id: number;
  mcc: string;
  mnc: string;
  iso: string;
  country: string;
  country_code: string;
  network: string;
  created_at: string;
  updated_at: string;
}

export interface MccMncCreateRequest {
  mcc: string;
  mnc: string;
  iso: string;
  country: string;
  country_code: string;
  network: string;
}

export interface MccMncUpdateRequest {
  mcc: string;
  mnc: string;
  iso: string;
  country: string;
  country_code: string;
  network: string;
}

export interface MccMncFilterOptions {
  countries: string[];
  networks: string[];
  isos: string[];
}

export interface MccMncListResponse {
  data: MccMnc[];
  pagination: {
    page: number;
    per_page: number;
    total: number;
    total_pages: number;
  };
}

export const mccMncService = {
  // Get all MCC-MNC records with pagination and filters
  getAll: async (params?: {
    search?: string;
    country?: string;
    network?: string;
    iso?: string;
    sort_by?: string;
    sort_order?: 'asc' | 'desc';
    page?: number;
    per_page?: number;
  }): Promise<MccMncListResponse> => {
    const searchParams = new URLSearchParams();
    
    if (params?.search) searchParams.append('search', params.search);
    if (params?.country && params.country !== 'all') searchParams.append('country', params.country);
    if (params?.network && params.network !== 'all') searchParams.append('network', params.network);
    if (params?.iso && params.iso !== 'all') searchParams.append('iso', params.iso);
    if (params?.sort_by) searchParams.append('sort_by', params.sort_by);
    if (params?.sort_order) searchParams.append('sort_order', params.sort_order);
    if (params?.page) searchParams.append('page', params.page.toString());
    if (params?.per_page) searchParams.append('per_page', params.per_page.toString());

    const response = await apiClient.get<MccMncListResponse>(`/mcc-mnc?${searchParams.toString()}`);
    return response;
  },

  // Get MCC-MNC record by ID
  getById: async (id: number): Promise<MccMnc> => {
    const response = await apiClient.get<MccMnc>(`/mcc-mnc/${id}`);
    return response;
  },

  // Create new MCC-MNC record
  create: async (data: MccMncCreateRequest): Promise<MccMnc> => {
    const response = await apiClient.post<MccMnc>('/mcc-mnc', data);
    return response;
  },

  // Update MCC-MNC record
  update: async (id: number, data: MccMncUpdateRequest): Promise<MccMnc> => {
    const response = await apiClient.put<MccMnc>(`/mcc-mnc/${id}`, data);
    return response;
  },

  // Delete MCC-MNC record
  delete: async (id: number): Promise<void> => {
    await apiClient.delete<void>(`/mcc-mnc/${id}`);
  },

  // Bulk delete MCC-MNC records
  bulkDeleteMccMnc: async (ids: number[]): Promise<{ message: string; count: number }> => {
    const response = await apiClient.delete<{ message: string; count: number }>(`/mcc-mnc/bulk-delete?ids=${ids.join(',')}`);
    return response;
  },

  // Get filter options for MCC-MNC
  getFilterOptions: async (): Promise<MccMncFilterOptions> => {
    const response = await apiClient.get<MccMncFilterOptions>('/mcc-mnc/filter-options');
    return response;
  },
}; 